### ISSUE TITLE
Reproducing Results in Paper

### ISSUE THREAD

**Issue by juntongshi48** [amazon-science/tabsyn#19]
Hi @hengruizhang98,

Thank you for your amazing work and for sharing such an easy-to-use code base.

I tried to reproduce the results of a few baseline models, but the results I got differed quite notably from those in the paper. For example, the blue italic entries in the table below show the result I got for the column-wise density metric (Tab 1&2 in the paper). 
![Screenshot 2024-06-18 at 18 05 24](https://github.com/amazon-science/tabsyn/assets/90075506/ff57135b-46c3-42f5-b94f-a2bf7e883b5a)
Generally, in the results I got, **the order of performance between the baselines was preserved, but the absolute magnitude of the performance got a bit worse.** The same phenomenon also happens on other metrics -- mle, alpha-precision, and beta-recall. 

Is this something wired or in the acceptable range of fluctuation?

Additionally, I followed the exact instruction in the README and downgraded the version of SDMetrics to 0.11.1 as suggested in #11.

Thank you so much :)

**Comment by hengruizhang98** [comment]
Hi Juntong,

Thanks for your interest! I just tried my code and obtained the following result on adult:
![image](https://github.com/amazon-science/tabsyn/assets/26669850/89172b1e-63e9-4fbf-8712-ff2f3c5582e6).

There is a minor bug in the code, and I believe you can obtain a similar result after fixing it:

In https://github.com/amazon-science/tabsyn/blob/6d3519249af2839932d6ee4985ca4d95f9dfe4a1/tabsyn/main.py#L87
It should be "best_loss = curr_loss".

Best,
Hengrui

**Comment by juntongshi48** [comment]
Hi Hengrui,

Thx for your speedy reply! This fix is magical, and I'm getting the similar results as you now. My questions are satisfied so I will close the issue.

Juntong
